transaction_apple_pay_details_integration_test.go

```diff
@@ -49,6 +49,9 @@ func TestTransactionApplePayDetails(t *testing.T) {
 	if tx.ApplePayDetails.CardholderName == "" {
 		t.Fatal("Expected ApplePayDetails to have CardholderName set")
 	}
+	if tx.ApplePayDetails.ProductID == "" {
+		t.Errorf("Expected ApplePayDetails to have ProductID set")
```

Comment:
I think it should be a warning

Reply:
I see that it is being set as "Unknown" when we don't send L2 and L3 data if that makes any difference

Reply:
I have changed it to LogF